### ISSUE TITLE
A way for empty string as value of variable

### ISSUE THREAD

**Issue by lihuanshuai** [Huachao/vscode-restclient#322]
empty string `''` is widely used in url parameters, is there any way to declare a variable to be empty string?

maybe something like

```
@q={{$empty}}
```

I am using a workaround now, config

```json
    "rest-client.environmentVariables": {
        "$shared": {
            "empty": ""
        }
    },
```

then `{{empty}}` could be used inside `.rest` file.

but it is not elegant enough, so a system variable may be necessary

**Comment by Huachao** [comment]
@lihuanshuai Can you provide me with more details about the usage of empty string?

**Comment by lihuanshuai** [comment]
@Huachao sorry for late reply. I am testing an api with different parameter groups. The code looks like:

```http
@param1 = test1
@param2 = test2

# @param1 = default
# @param2 = {{empty}}

GET https://{{host}}/api/items
        ?param1={{param1}}
        &param2={{param2}}
```

In this way, I could comment/uncomment some lines to switch parameters. yet, some parameter may be `''` as a default value, if there is no `{{empty}}`, there won't be a way to declare such variable.

You may say, comment/uncomment `&param2={{param2}}` line is an option, yet isn't it too complicated? (linter will complain too). So, I think a variable for empty string will be useful.

**Comment by Huachao** [comment]
@lihuanshuai thanks for your reply, in my view, the two methods provided from you, that a query parameter like `param2=` and without `param2` query parameter may be different for some server web frameworks. So if they act the same, they uri may look like `GET https://api.example.com?param1=`, right?

**Comment by lihuanshuai** [comment]
@Huachao right

**Comment by borekb** [comment]
We have an API that looks like this:

```
/products?filter=some-filter
```

When `some-filter` is empty, all products (unfiltered) are returned.

The proposed feature would allow for a `.http` file like this:

```txt
@filter =
# or @filter = {{EMPTY}} or something

GET {{apiroot}}/products?filter={{filter}}
```

This is currently invalid (and setting `@filter = ''` is a different thing).

**Comment by Huachao** [comment]
@borekb I am in favor of the proposed syntax without trailing space like Shell variables

**Comment by Huachao** [comment]
@lihuanshuai @borekb @pranayrauthu I have implemented this feature, and you can define empty file variables as following:
```http
@empty =
```
The trailing whitespaces are allowed, that means you can add some whitespaces after the equal sign, and the spaces will be removed in the evaluated result

**Comment by borekb** [comment]
Sounds great! Will test it out when a new version is out. Thank you.

**Comment by Huachao** [comment]
@lihuanshuai @borekb @pranayrauthu I have published this in the latest release 0.23.1 and you can verify it with the following syntax.

>  ```http
> @empty =
> ```
> The trailing whitespaces are allowed, that means you can add some whitespaces after the equal sign, and the spaces will be removed in the evaluated result
